src/data/finset/lattice.lean

```diff
@@ -119,6 +119,34 @@ begin
     { rw [finset.sup_insert, id.def], exact h h_subset.1 (h₁ htne h_subset.2), }, },
 end
 
+lemma sup_le_of_le_directed {α : Type*} [complete_lattice α] (s : set α)
+  (hdir : directed_on (≤) s) (t : finset α) :
+  t.nonempty → (∀ x ∈ t, ∃ y ∈ s, x ≤ y) → ∃ x, x ∈ s ∧ t.sup id ≤ x :=
```

Comment:
I think it might be preferable to assume `s.nonempty` and drop the `t.nonempty` condition.

Reply:
I like that. That simplifies this lemma a lot and `s.nonempty` is the case in the only place I apply it.